packages/frontend/src/features/sqlRunner/components/TablesPanel.tsx

```diff
@@ -1,56 +1,109 @@
-import { Box } from '@mantine/core';
+import { Box, LoadingOverlay, Text } from '@mantine/core';
+import { useEffect, useState } from 'react';
 import { Panel, PanelGroup, PanelResizeHandle } from 'react-resizable-panels';
 import { useAppSelector } from '../store/hooks';
 import { TableFields } from './TableFields';
 import { Tables } from './Tables';
 
-export const TablesPanel = () => {
+type TablesPanelProps = {
+    isLoading: boolean;
+    error: string | null;
+};
+
+export const TablesPanel: React.FC<TablesPanelProps> = ({
+    isLoading,
+    error,
+}) => {
     const initialPanelSizes = [80, 20];
     const activeTable = useAppSelector((state) => state.sqlRunner.activeTable);
+
+    // state for controlling the "still loading" message
+    const [showLoadingMessage, setShowLoadingMessage] = useState(false);
+
+    // effect to handle loading timeout
```

Comment:
This is great and works really well, but we usually try to use `mantine` hooks that we have already available to us so we don't have to worry about the timeout clearing on unmount as well 👍 

Here's an implementation suggestion:

```tsx
 // state for controlling the "still loading" message
    const [showLoadingMessage, setShowLoadingMessage] = useState(false);

    const { start, clear } = useTimeout(
        () => setShowLoadingMessage(true),
        3000,
    );

    useEffect(() => {
        if (isLoading) {
            setShowLoadingMessage(false);
            start();
        } else {
            clear();
            setShowLoadingMessage(false);
        }
    }, [isLoading, start, clear]);
```

Reply:
Nice. I need to get familiar with meantine hooks and utility funcs